sdk/core/core/src/az_http_policy.c

```diff
@@ -92,8 +96,35 @@ AZ_NODISCARD az_result az_http_pipeline_policy_logging(
     az_http_response * const response) {
   (void)data;
 
-  // Authentication logic
-  return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  if (data == NULL) {
+    return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  }
+
+  clock_t const threshold
+      = (data == NULL) ? 0 : ((az_log_options const *)data)->slow_response_threshold;
```

Comment:
data can't never be == NULL here.  when it is null, we are returning next policy in the line above

Reply:
You're right, I rewrote that part now.